src/Microsoft.Azure.WebJobs.Host/Indexers/DefaultTypeLocator.cs

```diff
@@ -122,10 +122,16 @@ private Type[] FindTypes(Assembly assembly, IEnumerable<Assembly> extensionAssem
             catch (ReflectionTypeLoadException ex)
             {
                 _log.WriteLine("Warning: Only got partial types from assembly: {0}", assembly.FullName);
+                _log.WriteLine($"The following loader failures occured when trying to load the assembly:");
+                string loaderFailuresMessage = string.Join(Environment.NewLine, ex.LoaderExceptions.Select(e => $"   - {e.Message}"));
+                _log.WriteLine(loaderFailuresMessage);
+
+                _log.WriteLine("This can occur if the assemblies listed above are missing, outdated or mismatched.");
+
                 _log.WriteLine("Exception message: {0}", ex.ToString());
 
                 // In case of a type load exception, at least get the types that did succeed in loading
-                types = ex.Types;
+                types = ex.Types.Where(t => t != null).ToArray();
```

Comment:
This is an additional change for safety. Types that fail to load will still be present in the array, but `null`, and while the existing code seems to cope with that fine, this increases safety.